spark-commons/src/test/scala/za/co/absa/spark/commons/implicits/StructTypeImplicitsTest.scala

```diff
@@ -216,4 +218,33 @@ class StructTypeImplicitsTest extends AnyFunSuite with SparkTestBase with JsonTe
     assert(schemaB.diffSchema(schemaA).isEmpty)
   }
 
+
+  test("isOfType should return true if the field type matches the expected type") {
+    val path = "/some/path"
+    val results = structTypeEnhancements.isOfType[StringType](path)
+
+    assert(results == true)
+  }
+
+  test("it should return false if the field type does not match the expected type") {
+    val path = "/another/path"
+    val results = structTypeEnhancements.isOfType[StringType](path)
+
+    assert(results == false)
+  }
+
+  test("it should return false if the path is invalid or does not exist") {
+    val path = "/nonexistent/path"
+    val results = structTypeEnhancements.isOfType[BooleanType](path)
+
+    assert(results == false)
+  }
+
+  test("it should return false if the field type is NullType") {
```

Comment:
If `NullType` is not used in the implementation I don't think this test case is needed. But can stay 😉

Reply:
Actually it works great as a negative example 👍